web/src/app.dev.js

```diff
@@ -0,0 +1,44 @@
+import 'babel-polyfill';
+
+import createHistory from 'history/createBrowserHistory';
+import React from 'react';
+import 'react-dates/initialize';
+import ReactDOM from 'react-dom';
+import { AppContainer } from 'react-hot-loader';
+import { routerMiddleware } from 'react-router-redux';
+import { createStore, applyMiddleware } from 'redux';
+import { createLogger } from 'redux-logger'; // eslint-disable-line import/no-extraneous-dependencies
+import thunk from 'redux-thunk';
+
+import { initI18n } from './i18n';
+import reducer from './reducers';
+import Root from './components/Root';
+
+import './styles/index.css';
+
+// Set locale based on browser language
+// if it matches one of our SUPPORTED_LOCALES
+// otherwise, set to DEFAULT_LOCALE ("en")
+initI18n();
+
+const history = createHistory();
+
+const middleware = [thunk, routerMiddleware(history), createLogger()];
+
+const store = createStore(reducer, applyMiddleware(...middleware));
+
+const render = (Component, props) => {
+  ReactDOM.render(
+    <AppContainer>
+      <Component {...props} />
+    </AppContainer>,
+    document.getElementById('app')
+  );
+};
+
+module.hot.accept('./components/Root.js', () => {
+  const HotRoot = require('./components/Root').default;
+  render(HotRoot, { history, store });
+});
+
+render(Root, { history, store });
```

Comment:
Create a separate entrypoint for dev just because that's less likely to go awry than using conditional logic in the main entrypoint.

Reply:
This is a super smart idea!